orchestrators/orchestrator.go

```diff
@@ -16,19 +19,30 @@ type Orchestrator interface {
 }
 
 // GetOrchestrator returns the Orchestrator as specified in configuration
-func GetOrchestrator(c *handler.Bivac) Orchestrator {
-	orch := c.Config.Orchestrator
-	log.Debugf("Using orchestrator: %s", orch)
-
-	switch orch {
-	case "docker":
-		return NewDockerOrchestrator(c)
-	case "kubernetes":
-		return NewKubernetesOrchestrator(c)
-	case "cattle":
-		return NewCattleOrchestrator(c)
+func GetOrchestrator(c *handler.Bivac) (orch Orchestrator, err error) {
+	if c.Config.Orchestrator != "" {
+		log.Debugf("Choosing orchestrator based on configuration...")
+		switch c.Config.Orchestrator {
+		case "docker":
+			orch = NewDockerOrchestrator(c)
+		case "kubernetes":
+			orch = NewKubernetesOrchestrator(c)
+		case "cattle":
+			orch = NewCattleOrchestrator(c)
+		default:
+			err = fmt.Errorf("'%s' is not a valid orchestrator", c.Config.Orchestrator)
+			return
+		}
+	} else {
+		log.Debugf("Detecting orchestrator based on environment...")
+		if detectCattle() {
```

Comment:
It may be better to detect kubernetes first in case rancher-metadata is still available in Rancher2.

Reply:
You're right, fixed by 2d8c6d6.